source/qb64pe.bas

```diff
@@ -24708,6 +24708,67 @@ SUB PreParse (e$)
     DIM f AS _FLOAT
     STATIC TotalPrefixedPP_TypeMod AS LONG, TotalPP_TypeMod AS LONG
 
+    DO 'convert &H values to decimal values to prevent errors
+        l = INSTR(l, UCASE$(e$), "&H")
+        IF l THEN
+            FOR l1 = l + 2 TO LEN(e$)
+                SELECT CASE UCASE$(MID$(e$, l1, 1))
+                    CASE "0" TO "9"
+                    CASE "A" TO "F"
+                    CASE ELSE:
+                        PRINT UCASE$(MID$(e$, l1, 1))
+                        EXIT FOR
+                END SELECT
+            NEXT
+            IF l1 <> l + 2 THEN 'hex number found
+                IF l1 > l + 18 THEN EXIT DO
```

Comment:
Should this return back an error instead?

(Same question about the other `EXIT DO` as well)

Reply:
It does.  Give it a try:   ``PRINT &H12345678901234567``

Overflow Error on Line ###.